File: packages/shared/src/types/category.ts
// Category — platform-agnostic representation
export interface Category {
  id: string;
  userId: string;
  name: string;
  color: string;
  icon: string;
  displayOrder: number | string; // DB returns string for numeric columns
  isArchived: boolean;
  createdAt: Date;
  updatedAt: Date;
}

// Available colors for categories
export const categoryColors = [
  "emerald",
  "indigo",
  "sky",
  "amber",
  "purple",
  "slate",
  "blue",
  "rose",
  "teal",
  "orange",
  "pink",
  "cyan",
] as const;

export type CategoryColor = (typeof categoryColors)[number];

// Available icons for categories (Lucide icon names)
export const categoryIcons = [
  "Sparkles",
  "SlidersHorizontal",
  "Mic2",
  "BookOpen",
  "Layers",
  "Circle",
  "Music",
  "Headphones",
  "Guitar",
  "Piano",
  "Drum",
  "Radio",
  "Video",
  "Camera",
  "Briefcase",
  "GraduationCap",
  "Users",
  "Calendar",
  "Star",
  "Heart",
  "Zap",
  "Trophy",
] as const;

export type CategoryIcon = (typeof categoryIcons)[number];

File: apps/ios/Seder/Models/Category.swift
import Foundation

struct Category: Codable, Identifiable {
    let id: String
    let name: String
    let color: String
    let icon: String
    let displayOrder: String?
    let isArchived: Bool
    let createdAt: String?
    let updatedAt: String?

    var order: Int { Int(displayOrder ?? "0") ?? 0 }
}

struct CreateCategoryRequest: Encodable {
    let name: String
    let color: String
    let icon: String
}

struct ReorderItem: Encodable {
    let id: String
    let displayOrder: Int
}
